File: HW2/Codes/Oz_Elifsena.lua
print("\nConditional") -- This aims to print numbers from 0 to 4 inclusive
for i = 0,4,1 do
    print("i: ", i)
end  

print("\nUnconditional")
i = 0
repeat --This aims to print numbers from 0 to 4 inclusive
    print("i: ", i)
    i = i + 1
    if (i == 4) then
        break --When i is 3 the loops is exited
    end
until Unconditional

print("\nUnlabeled")
for i = 0,3,1 do --This aims to print numbers from 0 to 2 inclusive
    print("i: ", i)
    for j = 3,4,1 do --This aims to print numbers from 3 to 4 inclusive
        if (j == 4) then
            break --When j is 4 the inner loop is exited
        end
        print("j: ", j)
    end
end

print("\nLabeled")
for i = 0,3,1 do --This aims to print numbers from 0 to 2 inclusive
    print("i: ", i)
    for j = 3,4,1 do --This aims to print numbers from 3 to 4 inclusive
        if (j == 4) then
            goto endLabel --When j is 4 the outer loop is exited by going to endlabel 
        end
        print("j: ", j)
    end
end
:: endLabel ::

File: HW2/Codes/Oz_Elifsena.php
<?php
   echo("\nConditional");
    for ($i = 0; $i <=4; $i++) { // This aims to print numbers from 0 to 4 inclusive
        echo("\ni: $i");
    }
    
    echo("\n\nUnconditional");
    for ($i = 0; $i <=4; $i++) { // This aims to print numbers from 0 to 4 inclusive
        echo("\ni: $i");
    
        if ($i == 3) {
            echo("\nExit loop");
            break; // When i is 3 the loops is exited
        }
    }
    
    echo("\n\nUnlabeled");
    for ($i = 0; $i < 3; $i++) { // This aims to print numbers from 0 to 2 inclusive
        echo("\ni: $i");
        for ($j = 3; $j <= 4; $j++) { // This aims to print numbers from 3 to 4 inclusive
            if ($j == 4) {
                break 1; // j is 4  // When j is 4 the inner loop is exited
            }
            echo("\nj: $j"); 
        }
    }
    
    echo("\n\nLabeled");
    for ($i = 0; $i < 3; $i++) { // This aims to print numbers from 0 to 2 inclusive
        echo("\ni: $i");
        for ($j = 3; $j <= 4; $j++) { // This aims to print numbers from 3 to 4 inclusive
            if ($j == 4) {
                goto labeledLoop; // j is 4 // When j is 4 the outer loop (labeledLoop) is exited
            }
            echo("\nj: $j"); 
        }
    }
    labeledLoop:
?>